test/extensions/request_id/uuid/config_test.cc

```diff
@@ -154,29 +161,45 @@ TEST(UUIDRequestIDExtensionTest, DISABLED_benchmark) {
 
 TEST(UUIDRequestIDExtensionTest, SetTraceStatus) {
   Random::RandomGeneratorImpl random;
-  UUIDRequestIDExtension uuid_utils(random);
-  TestRequestHeaderMapImpl request_headers;
+  UUIDRequestIDExtension uuid_utils(envoy::extensions::request_id::uuid::v3::UuidRequestIdConfig(),
+                                    random);
+  Http::TestRequestHeaderMapImpl request_headers;
   request_headers.setRequestId(random.uuid());
 
-  EXPECT_EQ(TraceStatus::NoTrace, uuid_utils.getTraceStatus(request_headers));
+  EXPECT_EQ(Tracing::Reason::NotTraceable, uuid_utils.getTraceReason(request_headers));
 
-  uuid_utils.setTraceStatus(request_headers, TraceStatus::Sampled);
-  EXPECT_EQ(TraceStatus::Sampled, uuid_utils.getTraceStatus(request_headers));
+  uuid_utils.setTraceReason(request_headers, Tracing::Reason::Sampling);
+  EXPECT_EQ(Tracing::Reason::Sampling, uuid_utils.getTraceReason(request_headers));
 
-  uuid_utils.setTraceStatus(request_headers, TraceStatus::Client);
-  EXPECT_EQ(TraceStatus::Client, uuid_utils.getTraceStatus(request_headers));
+  uuid_utils.setTraceReason(request_headers, Tracing::Reason::ClientForced);
+  EXPECT_EQ(Tracing::Reason::ClientForced, uuid_utils.getTraceReason(request_headers));
 
-  uuid_utils.setTraceStatus(request_headers, TraceStatus::Forced);
-  EXPECT_EQ(TraceStatus::Forced, uuid_utils.getTraceStatus(request_headers));
+  uuid_utils.setTraceReason(request_headers, Tracing::Reason::ServiceForced);
+  EXPECT_EQ(Tracing::Reason::ServiceForced, uuid_utils.getTraceReason(request_headers));
 
-  uuid_utils.setTraceStatus(request_headers, TraceStatus::NoTrace);
-  EXPECT_EQ(TraceStatus::NoTrace, uuid_utils.getTraceStatus(request_headers));
+  uuid_utils.setTraceReason(request_headers, Tracing::Reason::NotTraceable);
+  EXPECT_EQ(Tracing::Reason::NotTraceable, uuid_utils.getTraceReason(request_headers));
 
   // Invalid request ID.
   request_headers.setRequestId("");
-  uuid_utils.setTraceStatus(request_headers, TraceStatus::Forced);
+  uuid_utils.setTraceReason(request_headers, Tracing::Reason::ServiceForced);
   EXPECT_EQ(request_headers.getRequestIdValue(), "");
 }
 
-} // namespace Http
+TEST(UUIDRequestIDExtensionTest, SetTraceStatusPackingDisabled) {
+  Random::RandomGeneratorImpl random;
+  envoy::extensions::request_id::uuid::v3::UuidRequestIdConfig config;
+  config.mutable_pack_trace_reason()->set_value(false);
+  UUIDRequestIDExtension uuid_utils(config, random);
+
+  Http::TestRequestHeaderMapImpl request_headers;
+  request_headers.setRequestId(random.uuid());
+  EXPECT_EQ(Tracing::Reason::NotTraceable, uuid_utils.getTraceReason(request_headers));
+
+  uuid_utils.setTraceReason(request_headers, Tracing::Reason::Sampling);
+  EXPECT_EQ(Tracing::Reason::NotTraceable, uuid_utils.getTraceReason(request_headers));
```

Comment:
Might have missed it, but is there some test that validates the `modBy` value remains unchanged when packing is disabled?

Reply:
Will update SetTraceStatusPackingDisabled test to explicitly also check that the request ID has not changed.